### PR TITLE
executor: generalize SHOW TABLE t NEXT_ROW_ID for auto_random and sequences

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve?

There are no SQL statements to obtain the next SEQUENCE value, making it impossible to dump a SEQUENCE correctly (pingcap/dumpling#61).

### What is changed and how it works?

Generalized the existing `SHOW TABLE t NEXT_ROW_ID` / `ADMIN SHOW t NEXT_ROW_ID` to output the values of all ID allocators. 

### Check List 

Tests 

- Unit test

Side effects

- Breaking backward compatibility
    * One extra column for the result of `SHOW TABLE t NEXT_ROW_ID`

### Release note 

* The `ADMIN SHOW NEXT_ROW_ID` statement can now be used on sequences and tables with auto_random columns.

### CODE REVIEW

**Comment by kennytm** [comment]
**Note:** No `lastval(seq)` is not the proper solution, it returns NULL unless you have called `nextval(seq)` first, but that would change the sequence value.

<details><summary>Effect</summary>

```sql
mysql> create sequence aa;
Query OK, 0 rows affected (0.01 sec)

mysql> show table aa next_row_id;
+---------+------------+-------------+--------------------+----------------+
| DB_NAME | TABLE_NAME | COLUMN_NAME | NEXT_GLOBAL_ROW_ID | ID_TYPE        |
+---------+------------+-------------+--------------------+----------------+
| test    | aa         | _tidb_rowid |                  1 | AUTO_INCREMENT |
| test    | aa         |             |                  1 | SEQUENCE       |
+---------+------------+-------------+--------------------+----------------+
2 rows in set (0.00 sec)

mysql> select nextval(aa);
+-------------+
| nextval(aa) |
+-------------+
|           1 |
+-------------+
1 row in set (0.00 sec)

mysql> show table aa next_row_id;
+---------+------------+-------------+--------------------+----------------+
| DB_NAME | TABLE_NAME | COLUMN_NAME | NEXT_GLOBAL_ROW_ID | ID_TYPE        |
+---------+------------+-------------+--------------------+----------------+
| test    | aa         | _tidb_rowid |                  1 | AUTO_INCREMENT |
| test    | aa         |             |               1001 | SEQUENCE       |
+---------+------------+-------------+--------------------+----------------+
2 rows in set (0.00 sec)

mysql> create table cc (a int primary key auto_random);
Query OK, 0 rows affected, 1 warning (0.01 sec)

mysql> show table cc next_row_id;
+---------+------------+-------------+--------------------+----------------+
| DB_NAME | TABLE_NAME | COLUMN_NAME | NEXT_GLOBAL_ROW_ID | ID_TYPE        |
+---------+------------+-------------+--------------------+----------------+
| test    | cc         | _tidb_rowid |                  1 | AUTO_INCREMENT |
| test    | cc         | a           |                  1 | AUTO_RANDOM    |
+---------+------------+-------------+--------------------+----------------+
2 rows in set (0.00 sec)

mysql> insert into cc values ();
Query OK, 1 row affected (0.00 sec)

mysql> show table cc next_row_id;
+---------+------------+-------------+--------------------+----------------+
| DB_NAME | TABLE_NAME | COLUMN_NAME | NEXT_GLOBAL_ROW_ID | ID_TYPE        |
+---------+------------+-------------+--------------------+----------------+
| test    | cc         | _tidb_rowid |                  1 | AUTO_INCREMENT |
| test    | cc         | a           |              30001 | AUTO_RANDOM    |
+---------+------------+-------------+--------------------+----------------+
2 rows in set (0.00 sec)
```

</details>

**Known issue**: 

* The `RowIDAllocType` allocator is always present. Show this be removed when other allocators exist?

**Alternatives**: 

* In MariaDB the sequence is implemented using a table, so the value can be obtained from the table as well

    ```sql
    MariaDB [test]> show create sequence a;
    +-------+--------------------------------------------------------------------------------------------------------------------------+
    | Table | Create Table                                                                                                             |
    +-------+--------------------------------------------------------------------------------------------------------------------------+
    | a     | CREATE SEQUENCE `a` start with 1 minvalue 1 maxvalue 9223372036854775806 increment by 1 cache 1000 nocycle ENGINE=InnoDB |
    +-------+--------------------------------------------------------------------------------------------------------------------------+
    1 row in set (0.000 sec)

    MariaDB [test]> select * from a;
    +-----------------------+---------------+---------------------+-------------+-----------+------------+--------------+-------------+
    | next_not_cached_value | minimum_value | maximum_value       | start_value | increment | cache_size | cycle_option | cycle_count |
    +-----------------------+---------------+---------------------+-------------+-----------+------------+--------------+-------------+
    |                  2001 |             1 | 9223372036854775806 |           1 |         1 |       1000 |            0 |           0 |
    +-----------------------+---------------+---------------------+-------------+-----------+------------+--------------+-------------+
    1 row in set (0.000 sec)
    ```

* The same for PostgreSQL, where `select * from a` gives a table with columns `last_value`, `log_cnt` and `is_called`. Although one could also query the [`pg_sequence` table](https://www.postgresql.org/docs/10/catalog-pg-sequence.html).

File: executor/seqtest/seq_executor_test.go [hunk 1]
```diff
@@ -797,39 +797,76 @@ func HelperTestAdminShowNextID(c *C, s *seqTestSuite, str string) {
...
+	// Insert some rows.
+	tk.MustExec("insert into t3 (c) values (1), (2);")
+	r = tk.MustQuery(str + " t3 next_row_id")
+	r.Check(testkit.Rows("test1 t3 _tidb_rowid 1 AUTO_INCREMENT", "test1 t3 id 11 AUTO_RANDOM"))
+	// Rebase.
+	tk.MustExec("insert into t3 (id, c) values (103, 3);")
+	r = tk.MustQuery(str + " t3 next_row_id")
+	r.Check(testkit.Rows("test1 t3 _tidb_rowid 1 AUTO_INCREMENT", "test1 t3 id 114 AUTO_RANDOM"))
+
+	// test for a sequence.
```
Comment (AilinKid) [inline executor/seqtest/seq_executor_test.go:854]:
```suggestion
	// Test for a sequence.
```

File: executor/seqtest/seq_executor_test.go [hunk 2]
```diff
@@ -797,39 +797,76 @@ func HelperTestAdminShowNextID(c *C, s *seqTestSuite, str string) {
...
+
+	// test for a table with auto_random primary key
+	tk.MustExec("create table t3(id int primary key auto_random(5), c int)")
+	// Start handle is 1.
+	r = tk.MustQuery(str + " t3 next_row_id")
+	r.Check(testkit.Rows("test1 t3 _tidb_rowid 1 AUTO_INCREMENT", "test1 t3 id 1 AUTO_RANDOM"))
+	// Insert some rows.
+	tk.MustExec("insert into t3 (c) values (1), (2);")
+	r = tk.MustQuery(str + " t3 next_row_id")
+	r.Check(testkit.Rows("test1 t3 _tidb_rowid 1 AUTO_INCREMENT", "test1 t3 id 11 AUTO_RANDOM"))
```
Comment (AilinKid) [inline executor/seqtest/seq_executor_test.go:848]:
```suggestion
	r.Check(testkit.Rows("test1 t3 _tidb_rowid 1 AUTO_INCREMENT", "test1 t3 id **11** AUTO_RANDOM"))
```
Here may not be 11?
Also, the allocated step is dynamic for auto_random & rowid_type, assure it to be `30001` as what we saw in most of our test cases is not rigorous.

You can use fail-point to test it under a fixed-step for auto_random & rowid_type.

the case below is ditto~

Comment (AilinKid) [reply executor/seqtest/seq_executor_test.go:848]: Oh, I saw it at the beginning of this function~ sorry, it correct here.

Comment (kennytm) [reply executor/seqtest/seq_executor_test.go:848]: Line 792 should have fixed the step to 10. Are there more code to add?

```go
	step := int64(10)
	autoIDStep := autoid.GetStep()
	autoid.SetStep(step)
	defer autoid.SetStep(autoIDStep)
```

**Comment by AilinKid** [comment]
LGTM

**Comment by kennytm** [comment]
Ping @zimulala

**Pull request merged**